include/regs.h

```diff
@@ -4,13 +4,28 @@
 #ifndef REGS_H
 #define REGS_H
 
+#include "ultra64.h"
+
 #define REG_GROUPS 29 // number of REG groups, i.e. REG, SREG, OREG, etc.
 #define REG_PAGES 6
 #define REG_PER_PAGE 16
 #define REG_PER_GROUP REG_PAGES * REG_PER_PAGE
 
-/* We probably want a better name for gGameInfo based on OoT discussions */
-#define BASE_REG(n, r) gGameInfo->data[n * REG_PER_GROUP + r]
+typedef struct RegEditor {
+    /* 0x00 */ u8  regPage; // 0: no page selected (reg editor is not active); 1: first page; `REG_PAGES`: last page
+    /* 0x01 */ u8  regGroup; // Indexed from 0 to `REG_GROUPS`-1. Each group has its own character to identify it.
```

Comment:
Can a u8 be indexed from 0 to -1?

Reply:
It goes from `0` to `REG_GROUPS-1`. `REG_GROUPS` is 29 so it goes from `0` to `28`

Reply:
Ooooh, that makes a lot more sense 😆 thanks for the clarification